spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,185 @@
+
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
+
+        self._basis_type = basis_type
+        self._phase_convention = phase_convention
+
+        if normalization in ['sn3d', 'n3d', 'maxN']:
+            self._normalization = normalization
+        else:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN, but is {normalization}")
+
+        if channel_convention in ['acn', 'fuma']:
+            self._channel_convention = channel_convention
+        else:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {channel_convention}")
+
+        Signal.__init__(self, data, sampling_rate=sampling_rate,
+                        n_samples=n_samples, domain=domain, fft_norm=fft_norm,
+                        comment=comment, is_complex=is_complex)
+
+    @property
+    def n_max(self):
+        return self.n_max
+
+    @property
+    def basis_type(self):
+        return self._basis_type
+
+    @property
+    def normalization(self):
+        return self._normalization
+
+    @normalization.setter
+    def normalization(self, value):
+        if self.normalization is not value:
+            self._renormalize(self, value)
+
+        self._normalization = value
+
+    @property
+    def channel_convention(self):
+        return self._channel_convention
+
+    @channel_convention.setter
+    def channel_convention(self, value):
+        if self.channel_convention is not value:
+            self._change_channel_convention(value)
+
+    @property
+    def phase_convention(self):
+        return self._phase_convention
+
+    def _renormalize(self, normalization):
+        acn = range(0, (self.n_max + 1) ** 2)
+
+        if self.channel_convention == "fuma":
+            orders, degrees = fuma_to_nm(acn)
+        else:
+            orders, degrees = acn_to_nm(acn)
+
+        if self._normalization == 'n3d':
+            if normalization == "sn3d":
+                self._data[:, :, ...] *= \
+                    n3d_to_sn3d_norm(degrees, orders)
+            elif normalization == "maxN":
+                self._data[:, :, ...] *= \
+                    n3d_to_maxn(acn)
+
+        if self._normalization == 'sn3d':
+            # convert to sn3d
+            self._data[:, :, :] /= \
+                    n3d_to_sn3d_norm(degrees, orders)
+            if normalization == "maxN":
+                self._data[:, acn, :] *= n3d_to_maxn(acn)
+
+        if self._normalization == 'maxN':
+            # convert to n3d
+            self._data[:, acn, :] /= \
+                    n3d_to_maxn(acn)
+            if normalization == "sn3d":
+                self._data[:, acn, :] *= \
+                    n3d_to_sn3d_norm(acn)
+
+    def _change_channel_convention(self, value):
```

Comment:
This is nice as well, and I think we should have a shared (maybe public) function for that that the SH class uses as well.